statsmodels/regression/linear_model.py

```diff
@@ -330,6 +332,40 @@ def fit(self, method="pinv", cov_type='nonrobust', cov_kwds=None,
             # used in ANOVA
             self.effects = effects = np.dot(Q.T, self.wendog)
             beta = np.linalg.solve(R, effects)
+        elif method == "qr-pivot":
+            Q, R, P = linalg.qr(self.wexog, mode='economic', pivoting=True)
+            self.exog_Q, self.exog_R, self.exog_P = Q, R, P
+
+            # Cache singular values from R.
+            self.wexog_singular_values = np.linalg.svd(R, 0, 0)
+            if 'tol' in kwargs:
+                tol = kwargs['tol']
+            else:
+                tol = 1e-7
```

Comment:
My 2cent regarding numerical errors in computers. You may estimate the error (or in this case the tolerance) in the case that the numerical error is caused by rounding error of a sum of large number of elements.
It may not be trivial to estimate the errors in the case the errors are related to a singular value (like calculating the inverse of a rank deficient matrix)

It is common in many different tools (R, SAS …) to set the “tol” value as a parameter the user can set, and I’ve used the R defaults “tol = 1e-7”.

There is also some terminology issue and in this case the term “singular” refer to a mathematical point that is not defined (in the inverse matrix of X’X) where in Singular Value Decomposition the “Singular” is some scale parameter.

Reply:
> It is common in many different tools (R, SAS …) to set the “tol” value as a parameter the user can set, and I’ve used the R defaults “tol = 1e-7”.

It is OK for the user to set it, but the default should be set in a mathematically sound way. 1e-7 is arbitrary.